File: core/src/main/java/net/siegerpg/siege/core/drops/materials/CARROT.java
package net.siegerpg.siege.core.drops.materials;

import net.siegerpg.siege.core.drops.BlockDropTable;
import net.siegerpg.siege.core.drops.Reward;
import net.siegerpg.siege.core.items.implemented.misc.food.Carrot;
import net.siegerpg.siege.core.items.implemented.misc.food.Watermelon;
import net.siegerpg.siege.core.items.implemented.misc.materials.drops.blocks.Wheat;
import org.bukkit.Material;

public class CARROT extends BlockDropTable {
    public CARROT() {
        super(200, Material.CARROT, 0, 0, 0, 0, new Reward[]{
                new Reward(new Carrot(0).getUpdatedItem(false).asQuantity(1), 100.0),
                new Reward(new Carrot(0).getUpdatedItem(false).asQuantity(3), 10.0),
        });
    }
}

File: core/src/main/kotlin/net/siegerpg/siege/core/items/implemented/misc/food/Carrot.kt
package net.siegerpg.siege.core.items.implemented.misc.food

import net.siegerpg.siege.core.items.enums.Rarity
import net.siegerpg.siege.core.items.types.misc.CustomFood
import org.bukkit.Material
import org.bukkit.inventory.ItemStack

class Carrot() : CustomFood(
		name = "Carrot",
		customModelData = 330012,
		description = listOf("Baby carrots aren't", "a type of carrot"),
		levelRequirement = 0,
		material = Material.CARROT,
		health = 5.0,
                           ) {

	constructor(quality : Int) : this() {
		this.quality = 0
		this.rarity = Rarity.COMMON
		this.serialize()
	}

	constructor(item : ItemStack) : this() {
		this.item = item
		this.deserialize()
	}

}